cmd/docker/docker.go

```diff
@@ -55,7 +58,22 @@ func newDockerCommand(dockerCli *command.DockerCli) *cobra.Command {
 				return err
 			}
 
-			return plugincmd.Run()
+			err = plugincmd.Run()
+			if err != nil {
+				statusCode := 1
+				exitErr, ok := err.(*exec.ExitError)
+				if !ok {
+					return err
+				}
+				if ws, ok := exitErr.Sys().(syscall.WaitStatus); ok {
+					statusCode = ws.ExitStatus()
+				}
+				return cli.StatusError{
+					StatusCode: statusCode,
+					Status: fmt.Sprintf("Running plugin command `%s` errored.", strings.Join(args, " ")),
```

Comment:
Not sure if we shouldn't leave this blank instead. We could/should assume that the plugin printed some sort of error, I don't think a builtin command would print anything similarly generic on fail, would it?

Reply:
Agreed